src/lib/merkle_ledger_tests/test_mask.ml

```diff
@@ -156,477 +172,505 @@ module Make (Test : Test_intf) = struct
+            let num_accounts = 10 in
+            let account_ids = Account_id.gen_accounts num_accounts in
+            (* parent balances all non-zero *)
+            let balances =
+              List.init num_accounts ~f:(fun n ->
+                  Balance.of_nanomina_int_exn (n + 1) )
+            in
+            let parent_accounts =
+              List.map2_exn account_ids balances ~f:Account.create
+            in
+            (* add accounts to parent *)
+            List.iter parent_accounts ~f:(fun account ->
+                ignore @@ parent_create_new_account_exn maskable account ) ;
+            (* all accounts in parent to_list *)
+            let parent_list = Maskable.to_list_sequential maskable in
+            let zero_balance account =
+              Account.update_balance account Balance.zero
+            in
+            (* put same accounts in mask, but with zero balance *)
+            let mask_accounts = List.map parent_accounts ~f:zero_balance in
+            List.iter mask_accounts ~f:(fun account ->
+                ignore @@ create_existing_account_exn attached_mask account ) ;
+            let mask_list = Mask.Attached.to_list_sequential attached_mask in
+            (* same number of accounts after adding them to mask *)
+            assert (Stdlib.List.compare_lengths parent_list mask_list = 0) ;
+            (* should only see the zero balances in mask list *)
+            let is_in_same_order =
+              List.for_all2_exn parent_list mask_list
+                ~f:(fun parent_account mask_account ->
+                  Account_id.equal
+                    (Account.identifier parent_account)
+                    (Account.identifier mask_account) )
+            in
+            assert is_in_same_order ;
+            assert (
+              List.for_all mask_list ~f:(fun account ->
+                  Balance.equal (Account.balance account) Balance.zero ) ) ) )
+
+  let () =
+    add_test "masking in foldi" (fun () ->
+        Test.with_instances (fun maskable mask ->
+            let attached_mask = Maskable.register_mask maskable mask in
+            let num_accounts = 10 in
+            let account_ids = Account_id.gen_accounts num_accounts in
+            (* parent balances all non-zero *)
+            let balances =
+              List.init num_accounts ~f:(fun n ->
+                  Balance.of_nanomina_int_exn (n + 1) )
+            in
+            let parent_accounts =
+              List.map2_exn account_ids balances ~f:Account.create
+            in
+            (* add accounts to parent *)
+            List.iter parent_accounts ~f:(fun account ->
+                ignore @@ parent_create_new_account_exn maskable account ) ;
+            let balance_summer _addr accum acct =
+              accum + Balance.to_nanomina_int (Account.balance acct)
+            in
+            let parent_sum =
+              Maskable.foldi maskable ~init:0 ~f:balance_summer
```

Comment:
```suggestion
              Maskable.fold maskable ~init:0 ~f:balance_summer
```

and remove unused parameter from the `balance_summer` function (same modification could be applied below)

Reply:
unfortunately `fold` does not exist.